### PR TITLE
Less aggressive reschedule with unavailable conn of PG

### PR DESCRIPTION
**Pull request by jsc0218**
Related to #51277, when unavailable conn pops up for Postgres, aggressive schedule of removeOutdatedTables() delays normal threads reading within time constraint , as these threads compete with each other for locks. Because it is an unsuccessful conn, it might be unsuccessful again, why not reduce the frequence of removeOutdatedTables()?
 
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
make removeOutdatedTables() less aggressive with unsuccessful Postgres connection

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit b5a79b1bdadeb0695c02e8cf6e3999e1e683f941 with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/56609/b5a79b1bdadeb0695c02e8cf6e3999e1e683f941/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6831488907">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/56609/b5a79b1bdadeb0695c02e8cf6e3999e1e683f941/ci_running.html">❌ failure</a></td></tr>
<tbody>
</table>

File: src/Databases/PostgreSQL/DatabasePostgreSQL.cpp [hunk 1]
```diff
@@ -36,6 +36,7 @@ namespace ErrorCodes
 
 static const auto suffix = ".removed";
 static const auto cleaner_reschedule_ms = 60000;
+static const auto reschedule_error_multiplier = 10;
```
Comment (kssenii) [inline src/Databases/PostgreSQL/DatabasePostgreSQL.cpp:39]: let's just update `cleaner_reschedule_ms` to a bigger value

Comment (jsc0218) [reply src/Databases/PostgreSQL/DatabasePostgreSQL.cpp:39]: I think differentiate the normal case and unusual case is more reasonable. reschedule_error_multiplier is only used in the abnormal case.

Comment (kssenii) [reply src/Databases/PostgreSQL/DatabasePostgreSQL.cpp:39]: ok indeed

**Review by kssenii — APPROVED** [review]

**Comment by kssenii** [comment]
> AST fuzzer (ubsan)

https://github.com/ClickHouse/ClickHouse/issues/56661

> Integration tests (asan, analyzer) [3/6] — fail: 1

https://github.com/ClickHouse/ClickHouse/issues/56627

**Pull request merged**